pkgs/development/tools/misc/texlab/default.nix

```diff
@@ -3,32 +3,23 @@
 , fetchFromGitHub
 , nodejs
 , Security
-, texlab-citeproc-build-deps
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "texlab";
-  version = "1.7.0";
+  version = "1.9.0";
 
   src = fetchFromGitHub {
-    owner = "latex-lsp";
+    owner = "MetaDark";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "0b9lw6cmh7gyzj0pb3ghvqc3q7lzl12bfg9pjhl31lib3mmga8yb";
+    rev = "8e3be9b8372ad1ae63adfaf74a76294ef457e5d7";
+    sha256 = "1zis4v662bcck8qc5n0igfg5bhzl8glpjzbga92xgiz0nhj1ldn2";
   };
 
-  cargoSha256 = "0qnysl0ayc242dgvanqgmx8v4a2cjg0f1lhbyw16qjv61qcsx8y5";
-
-  nativeBuildInputs = [ nodejs ];
+  cargoSha256 = "06f9w13s2s4rrh7a1ik9pm9x2g2nwiw7z3k4qv1cr41ykasc4hcr";
 
   buildInputs = stdenv.lib.optionals stdenv.isDarwin [ Security ];
 
-  preBuild = ''
-    rm build.rs
-    ln -s ${texlab-citeproc-build-deps}/lib/node_modules/citeproc/node_modules src/citeproc/js
-    (cd src/citeproc/js && npm run dist)
-  '';
-
```

Comment:
Can't you just add a patch of https://github.com/cormacrelf/citeproc-rs/pull/39 to the derivation? I don't like that our update includes such a big diff which includes the removal of this preBuild and the node packages entries...

Reply:
I removed them because they aren't used anymore. The reason the build started to fail was because `citeproc-js` was replaced with `citeproc-rs`, see https://github.com/latex-lsp/texlab/commit/5a3e52a1f09c660dca467bf1aea200e9dad30297.

I'm not sure how to patch to dependencies of a rust package, so I'm just waiting for the changes to be fixed upstream. Then I will switch the derivation to use the main repo instead of my fork.

Reply:
Right... I see! so all good then. Now that your upstream version bump PR was merged you'll update the source right?

Reply:
Yep just testing the build right now :)